graphene/src/document.rs

```diff
@@ -119,13 +120,68 @@ impl Document {
 			.unwrap_or_default()
 	}
 
+	// Determines which layer is closer to the root, if path_a return true, if path_b return false
+	// Answers the question: Is A closer to the root than B?
+	pub fn layer_closer_to_root(&self, path_a: &Vec<u64>, path_b: &Vec<u64>) -> bool {
+		// Convert UUIDs to indices
+		let indices_for_path_a = self.indices_for_path(path_a).unwrap();
+		let indices_for_path_b = self.indices_for_path(path_b).unwrap();
+
+		let longest = max(indices_for_path_a.len(), indices_for_path_b.len());
+		for i in 0..longest {
+			// usize::MAX becomes negative one here, sneaky. So folders are compared as [X, -1]. This is intentional.
+			let index_a = *indices_for_path_a.get(i).unwrap_or(&usize::MAX) as i32;
+			let index_b = *indices_for_path_b.get(i).unwrap_or(&usize::MAX) as i32;
+
+			// index_a == index_b -> true, this means the "2" indices being compared are within the same folder
+			// eg -> [2, X] == [2, X] since we are only comparing the "2" in this iteration
+			// Continue onto comparing the X indices.
+			if index_a == index_b {
+				continue;
+			}
+
+			// If index_a is smaller, index_a is closer to the root
+			if index_a < index_b {
+				// path_a is closer
+				return true;
+			}
+
+			// If neither of the above are true, path_b is closer
+			break;
+		}
+
+		// path_b is closer
+		return false;
```

Comment:
I was a bit confused by all the loop manipulation.
coldnt lines 144-154 just be replaced by
```rust
return index_a < index_b
```
? But i guess it makes sense in terms of putting comments inbetween

Reply:
I think this simplification is a good suggestion - it is like this due to breaking down the problem into simplified pieces in an earlier stage. Reconsolidating it now makes sense!